src/OpenSage.Game/Network/SkirmishManager.cs

```diff
@@ -13,20 +13,61 @@ namespace OpenSage.Network
 {
     public abstract class SkirmishManager
     {
-        private string _connectionKey = string.Empty; // TODO: maybe use this for password protection?
+        protected static readonly NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();
 
         protected Game _game;
+
+        public SkirmishManager(Game game, bool isHosting)
+        {
+            _game = game;
+
+            IsHosting = isHosting;
+            Settings = new SkirmishGameSettings(isHosting);
+        }
+
+        public bool IsHosting { get; protected set; }
+        public SkirmishGameSettings Settings { get; protected set; }
+        public IConnection Connection { get; protected set; }
+
+        public abstract bool IsStartButtonEnabled();
+        public abstract Task HandleStartButtonClickAsync();
+
+        public virtual void Stop()
+        {
+        }
+    }
+
+    public sealed class LocalSkirmishManager : SkirmishManager
+    {
+        public LocalSkirmishManager(Game game)
+            : base(game, isHosting: true)
+        {
+            Settings.LocalSlotIndex = 0;
+            Settings.LocalSlot.State = SkirmishSlotState.Human;
+        }
+
+        public override bool IsStartButtonEnabled() => true;
+
+        public override Task HandleStartButtonClickAsync()
+        {
+            Connection = new EchoConnection();
+            Settings.Status = SkirmishGameStatus.ReadyToStart;
+            return Task.CompletedTask;
+        }
+    }
+
+    public abstract class NetworkSkirmishManager : SkirmishManager
```

Comment:
This is now the base class for the Host and Client skirmish managers (the new `LocalSkirmishManager` doesn't need the network stuff).